examples/platform/silabs/SiWx917/uart.cpp

```diff
@@ -16,69 +16,148 @@
+
+void Read_Capabilities(void)
 {
-    // UART init is already done
+    drv_capabilities = UARTdrv->GetCapabilities();
 }
 
-void USART_IRQHandler(void)
+void ARM_USART_SignalEvent(uint32_t event)
 {
+    switch (event)
+    {
+    case ARM_USART_EVENT_SEND_COMPLETE:
+        break;
+    case ARM_USART_EVENT_RECEIVE_COMPLETE:
 #ifdef ENABLE_CHIP_SHELL
-    chip::NotifyShellProcessFromISR();
-#endif
-#if defined(SL_WIFI)
-    /* TODO */
-#elif !defined(PW_RPC_ENABLED)
-    otSysEventSignalPending();
-#endif
-
-#if (defined(EFR32MG24) || defined(MGM24))
-    EUSART_IntClear(SL_UARTDRV_EUSART_VCOM_PERIPHERAL, EUSART_IF_RXFL);
-#endif
+        chip::NotifyShellProcessFromISR();
+#endif;
+        break;
+    case ARM_USART_EVENT_TRANSFER_COMPLETE:
+        break;
+    case ARM_USART_EVENT_TX_COMPLETE:
+        break;
+    case ARM_USART_EVENT_TX_UNDERFLOW:
+        break;
+    case ARM_USART_EVENT_RX_OVERFLOW:
+        break;
+    case ARM_USART_EVENT_RX_TIMEOUT:
+        break;
+    case ARM_USART_EVENT_RX_BREAK:
+        break;
+    case ARM_USART_EVENT_RX_FRAMING_ERROR:
+        break;
+    case ARM_USART_EVENT_RX_PARITY_ERROR:
+        break;
+    case ARM_USART_EVENT_CTS:
+        break;
+    case ARM_USART_EVENT_DSR:
+        break;
+    case ARM_USART_EVENT_DCD:
+        break;
+    case ARM_USART_EVENT_RI:
+        break;
+    }
 }
 
-/*
- *   @brief Callback triggered when a UARTDRV DMA buffer is full
- */
-static void UART_rx_callback(UARTDRV_Handle_t handle, Ecode_t transferStatus, uint8_t * data, UARTDRV_Count_t transferCount)
+void uartConsoleInit(void)
 {
-    (void) transferStatus;
+    int32_t status = 0;
+    Read_Capabilities();
 
-    uint8_t writeSize = (transferCount - lastCount);
-    if (RemainingSpace(&sReceiveFifo) >= writeSize)
+    status = UARTdrv->Initialize(ARM_USART_SignalEvent);
+    // Setting the GPIO 30 of the radio board (TX)
+    RSI_EGPIO_HostPadsGpioModeEnable(30);
+
+    // Initialized board UART
+    DEBUGINIT();
+    if (status != ARM_DRIVER_OK)
+    {
+        DEBUGOUT("\r\n UART Initialization Failed, Error Code : %d\r\n", status);
+    }
+    else
     {
-        WriteToFifo(&sReceiveFifo, data + lastCount, writeSize);
-        lastCount = 0;
+        DEBUGOUT("\r\n UART Initialization Success\r\n");
     }
 
-    UARTDRV_Receive(vcom_handle, data, transferCount, UART_rx_callback);
+    // Power up the UART peripheral
+    status = UARTdrv->PowerControl(ARM_POWER_FULL);
+    if (status != ARM_DRIVER_OK)
+    {
+        DEBUGOUT("\r\n Failed to Set Power to UART, Error Code : %d\r\n", status);
+    }
+    else
+    {
+        DEBUGOUT("\r\n Configured Power to UART \r\n");
+    }
 
-#ifdef ENABLE_CHIP_SHELL
-    chip::NotifyShellProcessFromISR();
-#endif
-#if defined(SL_WIFI)
-    /* TODO */
-#elif !defined(PW_RPC_ENABLED)
-    otSysEventSignalPending();
-#endif
+    // Enable Receiver and Transmitter lines
+    status = UARTdrv->Control(ARM_USART_CONTROL_TX, 1);
```

Comment:
What is the `1` in the UARTdrv->Control calls

Reply:
1 is the optional args, This is req to be 1 to enable the USART required flag